### PR TITLE
Api connection

### PR DESCRIPTION
**Pull request by Andyveloper**
# Project requirements

Read the [Bookstore API documentation](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739) to learn how to use the API.
- [x] Use [fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch) or [axios](https://axios-http.com/docs/intro) with [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) for making API calls.
- [x] Fetch book list from Bookstore API on Books page load.
- [x] Add an Action Creator, that returns a function.
- [x] Fetch list of books.
- [x] Dispatch another action (add an action type) that will pass API response to reducer and updates the state.
- [x] Refactor your add book and remove book features to persist your changes in the server.
- [x] Change your existing Action Creators, so they return functions.
- [x] Make changes by using API.
- [x] Dispatch an action (action type that existed before refactor) that will updates the state accordingly.
- [x] Add book and remove book should work in the same way after the refactor.
- [x] No styling is required.

### CODE REVIEW

File: src/components/Books/Book.js [hunk 1]
```diff
@@ -1,24 +1,31 @@
 import { CircularProgressbar } from 'react-circular-progressbar';
 import { PropTypes } from 'prop-types';
+import { useDispatch } from 'react-redux';
+import { deleteBookThunk } from '../../redux/apiConnection';
+// import { deleteBook } from '../../redux/books/books';
```
Comment (t-yanick) [inline src/components/Books/Book.js]: - [ ] Please kindly remove `commented code` from your `code base`, as it makes your `code base` dirty and less readable. To make sure that your codebase is always clean, always remember to remove commented-codes.

File: src/redux/apiConnection.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+export const postApiData = axios.post('https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/');
+
+export const currentApiKey = 'gwE0Bhw9RcasmYxC9anP';
+
+const createBookList = (book) => {
+  const bookList = [];
+  const key = Object.keys(book);
+  key.forEach((item) => {
+    // console.log(item);
```
Comment (t-yanick) [inline src/redux/apiConnection.js]: - [ ] Please kindly remove `commented code` from your `code base`, as it makes your `code base` dirty and less readable. To make sure that your codebase is always clean, always remember to remove commented-codes.

File: src/redux/books/books.js [hunk 3]
```diff
@@ -1,35 +1,30 @@
...
+      status: 'success',
+    }),
+    [getBooks.rejected]: (state) => ({
+      ...state,
+      status: 'error',
+    }),
+  },
 });
 
+// export const { addBook, deleteBook } = reducer.actions;
```
Comment (t-yanick) [inline src/redux/books/books.js]: - [ ] Please kindly remove `commented code` from your `code base`, as it makes your `code base` dirty and less readable. To make sure that your codebase is always clean, always remember to remove commented-codes.

**Review by t-yanick — CHANGES_REQUESTED** [review]
Hi @Zeraltz, 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Great job with RTK,
:heavy_check_mark: Your project is running as expected

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by roshan-bajgain — APPROVED** [review]
Hi @Zeraltz, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- No linter errors. ✔️
- GitHub flow is followed. ✔️
- Professional README file. ✔️
- PR has a good title and summary. ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**